torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1463,6 +1516,18 @@ def solve(self, disable_congruences=True, disable_equivalences=True):
         for source, expr in self._symbolic_equivalences:
             self._dynamic_results.add(f"{self._dcp.print_source(source)} == {self._dcp.doprint(expr)}")
 
+    @classmethod
+    def _is_supported_equivalence(cls, expr):
+        # Currently supported Dim ops are linear expressions with integer coefficients.
+        # So check that expr only contains +, *, ints, and a single occurrence of a symbol
```

Comment:
For future extenders, it would be helpful to explain why only linear expressions are supported

Reply:
Agreed.